packages/next-server/lib/head.tsx

```diff
@@ -8,6 +7,12 @@ type WithInAmpMode = {
   inAmpMode?: boolean
 }
 
+const headInstanceRefs: Set<React.RefObject<JSX.Element>> = new Set()
+
+const isServer = typeof window === 'undefined'
```

Comment:
It's actually better to write the check directly into the `if` statement because `typeof window` gets replaced at build time and tree shaking will be applied 💯

Reply:
Totally. This was mostly a transfer from the side-effect file. But I agree 100%

Reply:
That typeof replace is not standard (we recently added it) hence why I wanted to mention it in case you hadn't seen 💯 🚀

Reply:
👍